app/component/map/VehicleMarkerContainer.js

```
@@ -69,14 +77,19 @@ function shouldShowVehicle(message, direction, tripStart, pattern, headsign) {
 }
 
 function VehicleMarkerContainer(props, { config }) {
-  const visibleVehicles = Object.entries(props.vehicles).filter(([, message]) =>
-    shouldShowVehicle(
-      message,
-      props.direction,
-      props.tripStart,
-      props.pattern,
-      props.headsign,
-    ),
+  const visibleVehicles = Object.entries(props.vehicles).filter(
+    ([, message]) => {
+      const feed = message.route?.split(':')[0];
+      const ignoreHeadsign = config?.realTime?.[feed].ignoreHeadsign;
```

config and config.realtime are always available, no need for config?.realtime?. 

In general, code should not tolerate unexpected errors but fail fast and hard.